src/jemalloc.c

```diff
@@ -2024,6 +2025,25 @@ compute_size_with_overflow(bool may_overflow, dynamic_opts_t *dopts,
 	return true;
 }
 
+JEMALLOC_ALWAYS_INLINE void
+thread_event_prof_helper(tsd_t *tsd, size_t usize) {
+	thread_event(tsd, usize);
+	if (config_prof && opt_prof) {
```

Comment:
This should only happen when we set prof_sample_wait to 0; so let's add the "handle" function to each event type and do it there. Otherwise we end up 2 more branches on the medium path.